### DISCUSSION TITLE
On-disk permissions

### DISCUSSION THREAD

**Discussion by TremerePuck** [retronas/discussions#55] [Ideas]
I'm hosting a VM with RetroNAS on my Unraid server which works smoothly having files with nobody:users ownership. The problem is RetroNAS loves having user:user ownership.

Is there any way we could have a setting custom user:group permissions for files created and modified by RetroNAS and the Permissions fix tool?

Thank you!

**Comment by sairuk** [comment]
Yeh, I don't particularly see an issue with this providing this option, we could certainly entertain the fact. We use discussions for these new ideas while they are fleshed out though so I will convert this to a discussion

**Reply by sairuk** [reply]
We can probably approach this as an override option in the config file i suppose, I need to have a think about it because I also want to potentially change the config from yaml to json to support the wip webui properly. It'll be a longer term thing I think

**Reply by DanAE111** [reply]
I was actually going to bring up a the same discussion i have the same situation mapping the RetroNAS top level directory to a folder on my TrueNAS system via NFS. For now i've just modified the Ansible scripts which works but it might be worth discussing options.
- Adding an override for file and directory owner / group.
- Is there a different configuration of NFS that can be used to make the owner and group on the RetroNAS side appear as the local retronas user.
- Not officially supporting NFS and recommending a solution based on SMB.

Original post:
> Wanted to discuss this to see if there is perhaps a better way of doing what i've done here or if there could be some tweaks made to the installation scripts.
> 
> What i've done is setup a dataset on my TrueNAS system this dataset is then exposed to RetroNAS via NFS and a mapall user and group set on the TrueNAS side. This can then be mounted on the RetroNAS system without issue, files can be read and written as well as symlinks created.
> 
> The problem i run into is while running the installation scripts they fail as they are unable to change the owner and group of the files written to the NFS share. Modifying the included Ansible scripts to not set directory owner and group allows the install scripts to run successfully.
> 
> The directory structure appears as expected and with the correct permissions, symlinks are intact as expected.
> 
> My question now is have i gone about this the wrong way since the owner and group IDs on my NFS share do not match the retronas user on the RetroNAS system or is this a case where an advanced feature to allow bypassing the file and directory owner and group checks and modifications for items in RetroNAS top level directory be useful?

**Reply by sairuk** [reply]
We should have some kind of NFS solution/recommendation at some point.

The amount of work to add user:group support to the playbooks isn't small unfortunately but imo necessary. I might take a look at it in a separate branch soon and before we start supporting more stuff

**Reply by sairuk** [reply]
testing branch now has user/group split so you can map a local retronas user/group to whatever your nfs mount wants.

Should help with this.

**Reply by DanAE111** [reply]
Gave this a go and while the group setting did work it didn't solve my issue, i came to terms that it was an implementation issue on my part. What i ended up doing was mounting an SMB share instead and setting the UID and GID to match the local retronas user, so far so good.

Not really sure how you could do this without NFSv4 and ID mapping or just having RetroNAS ignore owner and group for files and folders, but at the end of the day that's an issue that has less to do with RetroNAS.

I'm happy using SMB also happy to contibute some information around this to the RetroNAS wiki.
